src/SUFST/src/can_unpack.c

```diff
@@ -151,5 +183,30 @@ void queue_receive_thread_entry(ULONG input)
         {
           return;
         }
+        /* For statistic */
+        tx_mutex_get(&unpack_ptr->stats.stats_mutex,TX_WAIT_FOREVER);
+        unpack_ptr->stats.tx_pdu_count++;
+        unpack_ptr->stats.txbytes += sizeof(pdu_struct);
+        tx_mutex_put(&unpack_ptr->stats.stats_mutex);    
     }
 }
+
+void stats_init(unpack_performance_t* stats)
+{
+    stats->rx_can_bps = 0;
+    stats->rx_can_count = 0;
+    stats->tx_pdu_bps = 0;
+    stats->tx_pdu_count = 0;
+    stats->rxbytes = 0;
+    stats->txbytes = 0;
+}
+
+void stats_timer_callback(unpack_performance_t* stats)
+{
+    stats->rx_can_bps = stats->rxbytes * 8 / STATS_TIMER_SECONDS; /* Amount of bits received per second */
```

Comment:
The mutex needs to be acquired here too so that the system timer thread will be suspended until the unpack thread has finished with the data.

Reply:
`8` should be a named constant. Like `const uint32_t BITS_PER_BYTE = 8;`

Another thought about these calculations, I always suggest making the operator ordering explicit with brackets (another MISRA C guideline). So it would be `(stats->rxbyxtes * 8) / STATS_TIMER_SECONDS;`